src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,7 +38,17 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        //this.value = address;
+        
+        this.addressComponents = address.split(", ");
+        if (!isValidAddress(this.addressComponents)) {
```

Comment:
It is better to merge the two validation methods. It doesn't make sense that it still could be invalid after the 1st validation test.